File: 18130.py
import sys


def main():
    n, q = (int(x) for x in sys.stdin.readline().split())
    ans, min_cost = -1, 10 ** 1000
    for i in range(1, n + 1):
        p, k, c = (int(x) for x in sys.stdin.readline().split())
        intervals = q // k
        if q % k == 0:
            intervals -= 1
        cost = p + intervals * (intervals + 1) // 2 * c
        if cost < min_cost:
            ans, min_cost = i, cost
    print(ans, min_cost)


main()

File: 18130.cpp
#include <iostream>

using namespace std;

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int n;
	long long q;
	cin >> n >> q;
	long long p, k, c;
	cin >> p >> k >> c;
	long long intervals = q / k;
	if(q % k == 0){
		--intervals;
	}
	int ans = 1;
	long long min_cost = p + intervals * (intervals + 1) / 2 * c;
	for(int i = 2; i <= n; ++i){
		cin >> p >> k >> c;
		intervals = q / k;
		if(q % k == 0){
			--intervals;
		}
		long long cost = p + intervals * (intervals + 1) / 2 * c;
		if(cost < min_cost){
			ans = i;
			min_cost = cost;
		}
	}
	cout << ans << ' ' << min_cost;
	return 0;
}
